caffe2/core/dispatch/CMakeLists.txt

```diff
@@ -8,6 +9,7 @@ set(TEST_SOURCES
 
 add_library(dispatch OBJECT ${LIB_SOURCES})
 target_enable_style_warnings(dispatch)
+target_include_directories(dispatch SYSTEM PUBLIC ${PROJECT_SOURCE_DIR}/third_party/flat_hash_map)
```

Comment:
Since we are not installing third_party/flat_hash_map, and this adds a target include directories that is public, it will cause issues during installation time. Either do installation of flat_hash_map or consider using a different one.

Reply:
cc @orionr for feedback on where I should put this file

Reply:
My advice is to vendor it as a "real" c10 header (rather than a submodule) that lives somewhere like `c10/FlatHashMap.h`